### DISCUSSION TITLE
Fetch ENS TXT records

### DISCUSSION THREAD

**Discussion by Madeindreams** [WalletConnect/walletconnect-monorepo#2371] [Developer Support] [answered]
There are 2 methods exposed from ethereumClient

fetchEnsName, fetchEnsAvatar

Is there a way to fetch other TXT record like the github and twitter?

**Comment by xzilja** [comment, marked as answer]
You shouldn't use methods from ethereumClient, these are for modal's internal use. As a dapp, just use wagmi ones, there are plenty documented at https://wagmi.sh Not sure if ens txt is there specifically, but you can always read from contracts directly.

**Reply by Madeindreams** [reply]
Thank you
